stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel_actions/ChannelActionsDialogFragment.kt

```diff
@@ -0,0 +1,146 @@
+package io.getstream.chat.android.ui.channel_actions
+
+import android.content.Context
+import android.os.Bundle
+import android.text.format.DateUtils
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.fragment.app.viewModels
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamFragmentChannelActionsBinding
+
+internal class ChannelActionsDialogFragment : BottomSheetDialogFragment() {
+    private val membersAdapter: ChannelMembersAdapter = ChannelMembersAdapter {
+        listener?.onMemberSelected(it)
+    }
+
+    private val cid: String by lazy { requireArguments().getString(ARG_CID)!! }
+    private val isGroup: Boolean by lazy { requireArguments().getBoolean(ARG_IS_GROUP, false) }
+
+    private var listener: OnChannelActionSelectedListener? = null
+
+    private val channelActionsViewModel: ChannelActionsViewModel by viewModels {
+        ChannelActionsViewModelFactory(cid, isGroup)
+    }
+
+    private var _binding: StreamFragmentChannelActionsBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        listener = (parentFragment ?: requireActivity()) as? OnChannelActionSelectedListener
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        listener = null
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = StreamFragmentChannelActionsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+        channelActionsViewModel.members.observe(viewLifecycleOwner) {
+            membersAdapter.submitList(it)
+            bindMemberNames(it)
+            bindMembersInfo(it)
+        }
+    }
+
+    override fun getTheme(): Int = R.style.StreamChannelActionsBottomSheetDialog
+
+    private fun setupViews() {
```

Comment:
Could be cleared up quite a bit with a `with(binding) { ... }`

Reply:
Done